finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -37,6 +37,13 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
   private[this] val closed: AtomicBoolean = new AtomicBoolean(false)
   protected[this] def isClosed = closed.get
 
+  private[this] val closedId: AtomicInteger = new AtomicInteger(0)
+  @tailrec private[this] def addClosedId(id: Int): Unit = {
+    val i = closedId.get
+    val max = if (id > i) id else i
+    if (!closedId.compareAndSet(i, max)) addClosedId(id)
+  }
```

Comment:
wouldn't this cause `t1` to try again with `id = 4` (and keep trying until successful)?